mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java

```diff
@@ -164,6 +165,18 @@ public int getPort() {
 
   public String getHostName() {
     maybeStart();
+    if (inetSocketAddress.getAddress() instanceof Inet6Address) {
+      Inet6Address address = (Inet6Address) inetSocketAddress.getAddress();
+      String hostName = inetSocketAddress.getHostName();
```

Comment:
It did happen for me in the test I'd written because it grabs the first non-loopback NIC it finds and then grabs the first local IP address. On my Linux machine it was using IPv6 and everything broke.

i.e.
inetSocketAddress.getHostName() returns "<ipv6 address>%2"

Calling MockWebServer.getUrl() exploded. Looks like scopes are generally not expected. HttpUrl throws IllegalArgumentException and if we got through that URL would throw MalformedURLException unless the % was escaped (and nothing expects the scope later either). Even whether the % should be escaped appears to be a bit controversial.

[I couldn't think of a better way of forcing a connection timeout as loopback addresses can't be "blackholed" with an unroutable address.]

As I said: not sure what to do about it and I agree a second PR once the way forward is clearer.